spring-pulsar-reactive/src/main/java/org/springframework/pulsar/reactive/core/ReactivePulsarTemplate.java

```diff
@@ -137,13 +140,14 @@ private Mono<MessageId> doSend(@Nullable String topic, T message, @Nullable Sche
 		// @formatter:on
 	}
 
-	private Flux<MessageId> doSendMany(@Nullable String topic, Flux<T> messages, @Nullable Schema<T> schema) {
+	private Flux<MessageSendResult<T>> doSendMany(@Nullable String topic, Flux<MessageSpec<T>> messages,
+			@Nullable Schema<T> schema) {
 		return messages.switchOnFirst((firstSignal, messageFlux) -> {
-			T firstMessage = firstSignal.get();
+			MessageSpec<T> firstMessage = firstSignal.get();
 			if (firstMessage != null && firstSignal.isOnNext()) {
-				String topicName = resolveTopic(topic, firstMessage.getClass());
-				ReactiveMessageSender<T> sender = createMessageSender(topicName, firstMessage, schema, null);
-				return messageFlux.map(MessageSpec::of).as(sender::sendMany)
+				String topicName = resolveTopic(topic, firstMessage.getValue().getClass());
```

Comment:
The good news is that with `MessageSpec` we can send tombstone/null values (it's not possible with a Flux<T> as a Flux can't emit null.
The bad news is that if the first message is a tombstone, we can't determine the topic or the schema (and this code will NPE 😬

Reply:
I like the good news better! 🤣 That is great news w/ the tombstone ability. 

I feel like we should add a null check on the value and log something accordingly? 

So basically we support tombstone now w/ send many w/ the one caveat that they can not be the 1st message in the published stream. Can we "switchOnSecond" in that case? I don't know if that is even a thing - just thinking of ideas Even if that worked, could they have 2 null values back to back at position 1 and 2?

Reply:
> Even if that worked, could they have 2 null values back to back at position 1 and 2?

They could have an infinite stream of nulls 😉

Reply:
Or a Flux that emits just one null MS and completes.

Reply:
BTW we have the same problem for the `send(T message)` operation.

Reply:
True and true. 
This PR is already above/beyond its initial goal. I have created https://github.com/spring-projects-experimental/spring-pulsar/issues/311 to follow up w/ the docs for sending tombstones.